### ISSUE TITLE
Provide a general use TaskExecutor for background tasks

### ISSUE THREAD

**Issue by philwebb** [spring-projects/spring-boot#5082]
Several parts of Spring Boot, and even more Spring Cloud could benefit from running tasks in the background or in parallel (see #5066 for example). It would be useful if we provided a common `TaskExecutor` so that threads could be shared.

**Comment by vpavic** [comment]
:+1:

**Comment by snicoll** [comment]
I have such a need on start.spring.io as well: I'd like to process events asynchronously and it would be nice if I could easily configure an `Executor` for that. Note that it's slightly different than this description. We would have to bind (somehow) that to `@EnableAsync`

**Comment by cemo** [comment]
:+1

**Comment by spencergibb** [comment]
Like @philwebb mentioned, spring-cloud has the need as well, see https://github.com/spring-cloud/spring-cloud-consul/issues/146

**Comment by sunlin901203** [comment]
what is the latest solution?

**Comment by sunlin901203** [comment]
I use @EnableAsync on Application class and use @Async on the medhod ,but it has no effect.

**Comment by wilkinsona** [comment]
@sunlin901203 As @snicoll already asked, please use Stack Overflow for this sort of question

**Comment by sunlin901203** [comment]
i only want to config a executor in application.properties. i do not want to use  @EnableAsync and  @Async and i do not want use `@Bean(name="taskExecutor")
public TaskExecutor taskExecutor() {
    ThreadPoolTaskExecutor taskExecutor = new ThreadPoolTaskExecutor();
    taskExecutor.setMaxPoolSize(this.poolSize);
    taskExecutor.setQueueCapacity(this.queueCapacity);
    taskExecutor.afterPropertiesSet();
    return new ThreadPoolTaskExecutor();
}`  too.i want config executor simple in application.properties like <task:executor id="taskExecutor" pool-size="20" queue-capacity="20"/>

**Comment by vpavic** [comment]
Any feedback on the current status of this issue, considering it was recently removed from the ```1.4.0.RC1```?

**Comment by wilkinsona** [comment]
It's something that we want to do, but it's not going to make it into 1.4

**Comment by vpavic** [comment]
Are there any chances we'll see this in Boot 2.0? Thanks.

**Comment by philwebb** [comment]
@vpavic Yeah, I'd love to see this in 2.0 but we won't get to it for a while. We've front loaded the 2.0 plan with all the reactive work and breaking changes (basically anything that we can't slip).

**Comment by joshiste** [comment]
As with Sprig 5.0.8 a new warning is logged, I guess more and more people would find a global AsyncTaskExecutor helpful (or at lease one for Spring MVC) which can be configured via properties.

```
2018-07-31 07:38:43.246  WARN 55362 --- [nio-8080-exec-9] o.s.w.s.m.m.a.ReactiveTypeHandler        :
!!!
Streaming through a reactive type requires an Executor to write to the response.
Please, configure a TaskExecutor in the MVC config under "async support".
The SimpleAsyncTaskExecutor currently in use is not suitable under load.
-------------------------------
Controller:	de.codecentric.boot.admin.server.web.ApplicationsController
Method:		applicationsStream
Returning:	reactor.core.publisher.Flux<org.springframework.http.codec.ServerSentEvent<de.codecentric.boot.admin.server.web.ApplicationsController$Application>>
!!!
```

**Comment by joshiste** [comment]
p.s. #1563 looks duplicate
